contentpool.install

```diff
@@ -42,6 +49,71 @@ function contentpool_install() {
   foreach ($source->listAll() as $config_name) {
     $config_storage->write($config_name, $source->read($config_name));
   }
+
+  contentpool_generate_keys();
+  contentpool_enable_cors();
+}
+
+/**
+ * Generates the OAuth Keys.
+ */
+function contentpool_generate_keys() {
+  // Build all the dependencies manually to avoid having to rely on the
+  // container to be ready.
+  $dir_name = 'keys';
+  /** @var \Drupal\simple_oauth\Service\KeyGeneratorService $key_gen */
+  $key_gen = \Drupal::service('simple_oauth.key.generator');
+  /** @var \Drupal\simple_oauth\Service\Filesystem\Filesystem $file_system */
+  $file_system = \Drupal::service('simple_oauth.filesystem');
+
+  $relative_path = DRUPAL_ROOT . '/../' . $dir_name;
+  if (!$file_system->isDirectory($relative_path)) {
+    $file_system->mkdir($relative_path);
+  }
+  $keys_path = $file_system->realpath($relative_path);
+  $pub_filename = sprintf('%s/public.key', $keys_path);
+  $pri_filename = sprintf('%s/private.key', $keys_path);
+
+  if ($file_system->fileExist($pub_filename) && $file_system->fileExist($pri_filename)) {
+    // 1. If the file already exists, then just set the correct permissions.
+    $file_system->chmod($pub_filename, 0600);
+    $file_system->chmod($pri_filename, 0600);
+  }
+  else {
+    // 2. Generate the pair in the selected directory.
+    try {
+      $key_gen->generateKeys($keys_path);
+    }
+    catch (\Exception $e) {
+      // Unable to generate files after all.
```

Comment:
that's not good, exception should go through and be visible so problems can be noted.